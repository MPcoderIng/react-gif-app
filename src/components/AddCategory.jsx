import { PropTypes } from "prop-types";
import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState([]);

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        // onAddCategory(categories => [inputValue, ...categories]);
        setInputValue('');
        onNewCategory(inputValue.trim());
        
    };

  return (
      <form onSubmit={ (event) => onSubmit(event) } aria-label="form">
          <input
              type="text"
              placeholder="Bucar Gifs"
              value={inputValue}
              onChange={(event) => onInputChange(event)}
          />
      </form>
 
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
