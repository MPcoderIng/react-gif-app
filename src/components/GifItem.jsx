import { PropTypes } from "prop-types"
export const GifItem = ({url, title}) => {
    // console.log(props.id);
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string.isRequired, 
  title: PropTypes.string.isRequired,
}

// AÃ±adir proptypes: title y url obligatorio
//evaluar snapshot
