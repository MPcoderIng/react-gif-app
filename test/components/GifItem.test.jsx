import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('PRUEBAS EN GIF ITEM', () => {
    
    const url = 'https://api.giphy.com/v2/emoji?api_key=PZdk0PbT1xkFEABpKXPfn7lIj1C495MF&limit=10&offset=0';
    const title = 'Titulo';

    test('should receive title and url', () => {
        render(<GifItem title={title} url={url}/>);
    });

    test('should match snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('should show correct url and alt', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
        // console.log(screen.getByRole('img').src);
        // screen.debug();
     });

     test('should show component title', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
      });
});