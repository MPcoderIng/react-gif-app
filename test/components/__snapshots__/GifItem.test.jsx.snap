// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRUEBAS EN GIF ITEM should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://api.giphy.com/v2/emoji?api_key=PZdk0PbT1xkFEABpKXPfn7lIj1C495MF&limit=10&offset=0"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
